### ISSUE TITLE
Magic header does not match the expected version.

### ISSUE THREAD

**Issue by ChristophKarlHeck** [pytorch/executorch#2533]
Hi guys,

I have integrated executorch into Mbed OS using an Arm Cortex M4 architecture (P-Nucleo-WB55RG). When I try to load the model, I get the following issue:
```
I executorch:main.cpp:68] Model PTE file loaded. Size: 960 bytes.
E executorch:program.cpp:108] Program identifier 'X��' != expected 'ET12'
I executorch:main.cpp:72] Program loading failed @ 0x0x20000E20: 0x23
F executorch:result.h:165] In function CheckOk(), assert failed: hasValue_
```

https://github.com/pytorch/executorch/blob/main/runtime/executor/program.cpp has caused the error in line 108.

How can I make sure that the magic header matches ET12?

Cheers,
Christoph

**Comment by mergennachin** [comment]
@ChristophKarlHeck 

This is related to backwards/forward compatibility with runtime and pte file.  We are still in alpha mode; the ExecuTorch APIs and the .pte binary format may change in incompatible ways before stabilizing in a future beta release. When deploying models, we currently recommend using a version of the runtime built from the same git revision that was used to generate the .pte file.

See our disclaimer in our documentation landing page: https://pytorch.org/executorch/stable/index.html

**Comment by AIWintermuteAI** [comment]
Oh, to be clear, since both pte and runtime are built with [this script](https://github.com/pytorch/executorch/blob/main/examples/arm/run.sh)
from this commit
```
commit ffad82455f91059f936bbd12803f4e5f3a537bf5 (HEAD -> main, origin/main, origin/gh/fredrikk/test, origin/HEAD)
Author: varunchariArm <150060871+varunchariArm@users.noreply.github.com>
Date:   Tue Oct 29 20:28:22 2024 -0700

    Update executorch-arm-delegate-tutorial.md with CS320 FVP
    
    Differential Revision: D65067462
    
    Pull Request resolved: https://github.com/pytorch/executorch/pull/6230
```
they should match, shouldn't they?

**Comment by ChristophKarlHeck** [comment]
Hi @AIWintermuteAI,
yes, I solved the issue in the meantime. It's been a while ago, and if I remember it correctly, than that patch has solved it for me:
https://github.com/ChristophKarlHeck/mbed-torch-fusion-os/blob/main/libs/utils/patches/pte_to_header.patch
Let me know if that helps!
Cheers,
Christoph

**Comment by AIWintermuteAI** [comment]
@ChristophKarlHeck that helped, thanks! I still face errors, but not from missing magic header.

**Comment by AIWintermuteAI** [comment]
It's missing add op, at the moment
```
I [executorch:arm_executor_runner.cpp:388] Model PTE file loaded. Size: 944 bytes.
I [executorch:arm_executor_runner.cpp:398] Model buffer loaded, has 1 methods
I [executorch:arm_executor_runner.cpp:406] Running method forward
I [executorch:arm_executor_runner.cpp:417] Setup Method allocator pool. Size: 131072 bytes.
I [executorch:arm_executor_runner.cpp:434] Setting up planned buffer 0, size 64.
E [executorch:operator_registry.cpp:185] kernel 'aten::add.out' not found.
E [executorch:operator_registry.cpp:186] dtype: 3 | dim order: [
E [executorch:operator_registry.cpp:186] 0,
E [executorch:operator_registry.cpp:186] ]
E [executorch:operator_registry.cpp:186] dtype: 3 | dim order: [
E [executorch:operator_registry.cpp:186] 0,
E [executorch:operator_registry.cpp:186] ]
E [executorch:operator_registry.cpp:186] dtype: 3 | dim order: [
E [executorch:operator_registry.cpp:186] 0,
E [executorch:operator_registry.cpp:186] ]
E [executorch:operator_registry.cpp:186] dtype: 3 | dim order: [
E [executorch:operator_registry.cpp:186] 0,
E [executorch:operator_registry.cpp:186] ]
```
I'm trying to run a graph with this single OP for sanity check.

**Comment by AIWintermuteAI** [comment]
Has something to do with DEXECUTORCH_SELECT_OPS_LIST it seems.

**Comment by AIWintermuteAI** [comment]
Mmm, weird, looks like I have no kernels?
Both softmax and add give me 
```
E [executorch:operator_registry.cpp:185] kernel 'aten::_softmax.out' not found.
```

**Comment by ChristophKarlHeck** [comment]
Have a look in this script:
https://github.com/ChristophKarlHeck/mbed-torch-fusion-os/blob/main/libs/utils/scripts/build_et_libs.sh
